### PR TITLE
Added bitrate selection

### PR DESCRIPTION
**Pull request by abhay-s-rawat**
Added bitrate functionality.
old request is below, please discard old one as it has some changes related to flutter 3 but new one has only changes related to bitrate functionality.
https://github.com/SimformSolutionsPvtLtd/audio_waveforms/pull/53

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## [0.1.5] - 19 Aug, 2022
+
+* Added bitRate feature for Android (Default to 32kbps).
+* Added bitRate feature for IOS (Default to 32kbps) (Not Tested).
```
Comment (ujas-m-simformsolutions) [inline CHANGELOG.md:4]: Combine these changes into just one line and remove `Not Tested`.

File: android/src/main/kotlin/com/simform/audio_waveforms/AudioWaveformsPlugin.kt [hunk 2]
```diff
@@ -31,6 +31,7 @@ class AudioWaveformsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
     private var encoder: Int = 0
     private var outputFormat: Int = 0
     private var sampleRate: Int = 16000
+    private var bitRate: Int = 32768
```
Comment (ujas-m-simformsolutions) [inline android/src/main/kotlin/com/simform/audio_waveforms/AudioWaveformsPlugin.kt:34]: I think 32 kbps is too low quality, we should default to 128 kbps.

File: example/lib/main.dart [hunk 3]
```diff
@@ -306,7 +307,11 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
     if (isRecording) {
       recorderController.reset();
       final path = await recorderController.stop(false);
-      if (path != null) await playerController5.preparePlayer(path);
+
+      if (path != null) {
+        print("Recorded file size: ${File(path).lengthSync()}");
```
Comment (ujas-m-simformsolutions) [inline example/lib/main.dart:312]: use `debugPrint` here.

File: ios/Classes/AudioRecorder.swift [hunk 4]
```diff
@@ -7,8 +7,9 @@ public class AudioRecorder: NSObject, AVAudioRecorderDelegate{
     var hasPermission: Bool = false
     public var meteringLevels: [Float]?
     
-    public func startRecording(_ result: @escaping FlutterResult,_ path: String?,_ encoder : Int?,_ sampleRate : Int?,_ fileNameFormat: String){
+    public func startRecording(_ result: @escaping FlutterResult,_ path: String?,_ encoder : Int?,_ sampleRate : Int?,_ bitRate : Int?,_ fileNameFormat: String){
         let settings = [
+            AVEncoderBitRateKey: bitRate ?? 32768,
```
Comment (ujas-m-simformsolutions) [inline ios/Classes/AudioRecorder.swift:12]: Same for ios, make it 128 kbps.

**Review by ujas-m-simformsolutions — CHANGES_REQUESTED** [review]

**Comment by abhay-s-rawat** [comment]
Will do the changes. You will find this funny but previously the bitrate was near 8kbps which is default. lol.
Are you sure bumping it to 128kbps will not have impact on already using users of this package , because after doing above changes the recorded file will be 16 times greater in size than before ?

**Comment by abhay-s-rawat** [comment]
128kbps is music level quality which user does not want in normal audios. That's why did 32kbps. Do you want me to proceed with 128kbps default ?

**Comment by ujas-m-simformsolutions** [comment]
@abhay-s-rawat Can you please share how you got the 8 kbps default value?

**Comment by ujas-m-simformsolutions** [comment]
And now I think, we should just let the platform decide if bitrate is not provided so lets make it `nullable` so if not provided then we won't set anything.

**Comment by abhay-s-rawat** [comment]
Letting the platform decide will only make things not right, because different platforms have different defaults, it will affect cross platform apps that currently are using this package.
Actually I caculated the audio file size and length and got somewhere near 8kbps so I got that values. I am not aware of platform defaults actually.
I suggest not keeping it too low and not too high. 32kbps/64kbs/96kbps will be best fit. What do you suggest ?

**Comment by ujas-m-simformsolutions** [comment]
@abhay-s-rawat We can go with the 64 kbps

**Comment by abhay-s-rawat** [comment]
Sure will do it, sorry for the delay , was but busy somewhere else . Will do changes asap.Thanks for your time

**Comment by abhay-s-rawat** [comment]
Did above changes as discussed. Sorry for the delay .

**Comment by ujas-m-simformsolutions** [comment]
@abhay-s-rawat Thank you for your contribution. This will be available in the next release.

**Pull request merged**